### ISSUE TITLE
`enum_variant_names` shouldn't trigger when all prefixes are `_`

### ISSUE THREAD

**Issue by c410-f3r** [rust-lang/rust-clippy#9018]
`_` is a special case contrary to any other user-defined prefix.

```rust
enum Enum {
    _Bar,
    _Foo,
}
```
